examples/hosts/app-integration/external-data/README.md

```diff
@@ -6,15 +6,54 @@ Please note that the ideas explored here are experimental and under development.
+
+This repo contains an external service that serves as a mock external "source of truth" data server, that offers this REST API collection and webhook interfaces in ./src/mock-external-data-service. The APIs served by this external service are the following:
+
+1. POST `/register-for-webhook`
+2. GET `/fetch-tasks`
+3. POST `/set-tasks`
+4. POST `/debug-reset-task-list`
+
+Find the documentation for them in the code itself: (./src/mock-external-data-service/service.ts)[./src/mock-external-data-service/service.ts]
+
+Next we need a service that will register the webhooks and listen for incoming changes. In a true implementation, this registration would happen in the Fluid server-side, potentially in the Alfred service, as we have in our dev branch here: https://github.com/microsoft/FluidFramework/blob/dev/external-data-prototyping/server/routerlicious/packages/lambdas/src/alfred/index.ts#L463-L508.
+
+However, for the purposes of this example, we have built out a separate service to register for the webhook. It contains the following endpoints:
+1. POST `/register-for-webhook`
+2. POST `/echo-external-data-webhook`
+
+Find the details in the code here: (examples/hosts/app-integration/external-data/src/mock-customer-service/service.ts)[examples/hosts/app-integration/external-data/src/mock-customer-service/service.ts]
+
+Next we come to how the Fluid collaboration session and clients will consume the APIs above and render them to the screen. To accomplish that, this example repo explores two routes to implement the scenario above. One is the Echo Webhook Pattern and the other is the Bot Pattern. Both are documented in more detail below.
 
 ### Echo Webhook Pattern
 
-TODO: Document the echo-webhook pattern and how it is surfaced in the example code.
+<img width="1356" alt="image" src="https://user-images.githubusercontent.com/6777404/216415477-14d0b193-29c9-48e5-8b6b-0a549a5dde58.png">
+
+<img width="1404" alt="image" src="https://user-images.githubusercontent.com/6777404/216417448-2a43db3e-12a2-48a6-b6d0-a6c4a95e27d1.png">
+
+<img width="1374" alt="image" src="https://user-images.githubusercontent.com/6777404/216417779-12861504-7909-489c-b7a2-4d75814a396f.png">
+
+In this pattern. the indivdual clients are responsible for authenticatin with the external data source and making REST calls as needed. However, since the the "customer server" from the section prior, is registered to the webhooks and listening for incoming changes, the clients do not have a way to know that there has been a change upstream. The "echo webhook" pattern is therefore a signal that the server side broadcasts to the Fluid service to echo the information that it has received from the webhook that there is incoming information.
```

Comment:
```suggestion
In this pattern. the individual clients are responsible for authentication with the external data source and making REST calls as needed. However, since the "customer server" from the section prior is registered to the webhooks and listening for incoming changes, the clients do not have a way to know that there has been a change upstream. The "echo webhook" pattern is therefore a signal that the server side broadcasts to the Fluid service to echo the information that it has received from the webhook that there is incoming information.
```